src/lib/features/metrics/unknown-flags/unknown-flags-service.ts

```diff
@@ -80,24 +81,9 @@ export class UnknownFlagsService {
         this.unknownFlagsCache.clear();
     }
 
-    async getGroupedUnknownFlags(): Promise<
-        { name: string; reportedBy: { appName: string; seenAt: Date }[] }[]
-    > {
-        const unknownFlags = await this.unknownFlagsStore.getAll();
-
-        const grouped = new Map<string, { appName: string; seenAt: Date }[]>();
-
-        for (const { name, appName, seenAt } of unknownFlags) {
-            if (!grouped.has(name)) {
-                grouped.set(name, []);
-            }
-            grouped.get(name)!.push({ appName, seenAt });
-        }
-
-        return Array.from(grouped.entries()).map(([name, reportedBy]) => ({
-            name,
-            reportedBy,
-        }));
+    async getAll(): Promise<UnknownFlag[]> {
+        if (!this.flagResolver.isEnabled('reportUnknownFlags')) return [];
+        return this.unknownFlagsStore.getAll();
```

Comment:
I think Ivar's idea here was to do something like `this.unknownFlagsStore.getAllAfter("7 days ago");` and keep all the flags in the database. Doing an insert and on conflict merge the date when it was seen.

That way we can keep the clean up process running once a day/week and deleting flags with more than 7 days

Reply:
I think that would simplify our logic, but it feels worse in terms of being defensive:
 - Unknown flags always hit the DB right away instead of being buffered in-memory per pod and only flushed/merged in a scheduled manner
 - In those 7 days (or whatever cadence we decide) we can be quite destructive by spamming Unleash with a huge volume of unknown flags

But maybe this is how our metrics work currently and we're okay with it 🤷
Let's chat about it.